scaleway/resource_storage_object_bucket.go

```diff
@@ -0,0 +1,108 @@
+package scaleway
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceScalewayStorageObjectBucket() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayStorageObjectBucketCreate,
+		Read:   resourceScalewayStorageObjectBucketRead,
+		Update: resourceScalewayStorageObjectBucketUpdate,
+		Delete: resourceScalewayStorageObjectBucketDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the bucket",
+			},
+			"acl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "private",
+				Description: "ACL of the bucket: either 'public-read' or 'private'. Private by default.",
+			},
+		},
+	}
+}
+
+func resourceScalewayStorageObjectBucketCreate(d *schema.ResourceData, m interface{}) error {
+	bucketName := d.Get("name").(string)
+	acl := d.Get("acl").(string)
+
+	s3Client := m.(*Meta).s3Client
+
+	_, err := s3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucketName),
+		ACL:    aws.String(acl),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(bucketName)
+
+	return resourceScalewayStorageObjectBucketRead(d, m)
+}
+
+func resourceScalewayStorageObjectBucketRead(d *schema.ResourceData, m interface{}) error {
+	bucketName := d.Get("name").(string)
+
+	s3Client := m.(*Meta).s3Client
+
+	_, err := s3Client.ListObjects(&s3.ListObjectsInput{
+		Bucket: aws.String(bucketName),
+	})
+	if err != nil {
+		if serr, ok := err.(awserr.Error); ok && serr.Code() == s3.ErrCodeNoSuchBucket {
+			log.Printf("[ERROR] Bucket %q was not found - removing from state!", d.Get("name").(string))
```

Comment:
The string cast is useless.

Reply:
Please use [this logger](https://github.com/terraform-providers/terraform-provider-scaleway/blob/master/scaleway/logger.go#L17).